src/aggregators/coevolve.jl

```diff
@@ -156,61 +266,59 @@ end
     @inbounds return p.rates[lidx](u, params, t)
 end
 
-function next_time(p::CoevolveJumpAggregation{T}, u, params, t, i, tstop::T) where {T}
-    @unpack rng, haslratevec = p
-    num_majumps = get_num_majumps(p.ma_jumps)
-    num_cjumps = length(p.urates) - length(p.rates)
+function next_time(p::CoevolveJumpAggregation, u, params, t, i)
+    @unpack next_jump, cur_rates, ma_jumps, rates, rng, pq, urates = p
+    num_majumps = get_num_majumps(ma_jumps)
+    num_cjumps = length(urates) - length(rates)
     uidx = i - num_majumps
     lidx = uidx - num_cjumps
     urate = uidx > 0 ? get_urate(p, uidx, u, params, t) : get_ma_urate(p, i, u, params, t)
-    last_urate = p.cur_rates[i]
-    if i != p.next_jump && last_urate > zero(t)
-        s = urate == zero(t) ? typemax(t) : last_urate / urate * (p.pq[i] - t)
-    else
-        s = urate == zero(t) ? typemax(t) : randexp(rng) / urate
+    if urate < zero(t)
+        error("urate = $(urate) < 0 for jump = $(i) at t = $(t) which is not allowed.")
     end
```

Comment:
Again, maybe we can drop this and assume users won't provide rate functions that give non-nonsensical output? (This is safer, but trades off performance.)

Reply:
Agreed.